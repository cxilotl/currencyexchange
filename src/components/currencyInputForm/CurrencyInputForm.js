import React from "react";
import PropTypes from 'prop-types';
import cssStyles from "./CurrencyInputForm.module.scss";
import amountValidator from './amountValidation';

const CurrencyInputForm = ({ currency, onChange }) => {
  const onInput = (e) => {
    const amount = parseFloat(e.currentTarget.value);
    if (amountValidator.validate(amount)) {
      onChange( e.currentTarget.value );
    } else {
      e.preventDefault();
    }
  };
  return(
    <form className={ cssStyles.form }>
      <label htmlFor="currentCurrency" className={ cssStyles.label }>
        { currency.toUpperCase() }
        <input
          id="currentCurrency"
          data-testid="currency-input"
          name="currentCurrency"
          className={ cssStyles.input }
          type="number"
          onChange={ onInput }
          placeholder="0.00"
          pattern="/^-?[0-9]+(\.[0-9][0-9]?)?$/g"
        />
      </label>
    </form>
  );
};

CurrencyInputForm.propTypes = {
  currency: PropTypes.string.isRequired,
  onChange: PropTypes.func
};

export default CurrencyInputForm;