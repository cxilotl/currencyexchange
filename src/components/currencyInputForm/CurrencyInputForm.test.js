import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import CurrencyInputForm from "./CurrencyInputForm";

describe('CurrencyInputForm', () => {
  const currency = 'gbp';

  it('Snapshot', () => {
    const component = renderer.create(
      <CurrencyInputForm
        currency={ currency }
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Given a CurrencyInputForm element, it', () => {
    afterEach(cleanup);

    it('should include a currency label', () => {
      const { getByLabelText } = render(
        <CurrencyInputForm currency={ currency }/>
      );
      const currencyLabel = getByLabelText('GBP');
      expect(currencyLabel).toBeInTheDocument();
    });

    it('should include an input for the currency value to exchange', () => {
      const { getByTestId } = render(
        <CurrencyInputForm currency={ currency }/>
      );
      const currencyInput = getByTestId('currency-input');
      expect(currencyInput).toBeInTheDocument();
    });

    // TODO: Test inout validation as well for
    // contain two inputs on the active exchange screen for both pockets. Each input should be validated to let to type only numbers with two digits after the dot
  });
});