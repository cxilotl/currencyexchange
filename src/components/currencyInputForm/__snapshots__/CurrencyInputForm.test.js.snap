// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInputForm Snapshot 1`] = `
<form
  className="form"
>
  <label
    className="label"
    htmlFor="currentCurrency"
  >
    GBP
    <input
      className="input"
      data-testid="currency-input"
      id="currentCurrency"
      name="currentCurrency"
      onChange={[Function]}
      pattern="/^-?[0-9]+(\\\\.[0-9][0-9]?)?$/g"
      placeholder="0.00"
      type="number"
    />
  </label>
</form>
`;
